imap_processing/codice/codice_l1a.py

```diff
@@ -343,86 +457,57 @@ def process_codice_l1a(file_path: Path, data_version: str) -> xr.Dataset:
-
-            elif apid in apids_for_lo_science_processing:
-                # Sort the packets by time
-                packets = sort_by_time(grouped_data[apid], "SHCOARSE")
-
-                # Determine the start time of the packet
-                met = packets[0].data["ACQ_START_SECONDS"].raw_value
-                met = [met, met + 1]  # TODO: Remove after SIT-3
-                # Extract the data
-                science_values = packets[0].data["DATA"].raw_value
-
-                # Get the four "main" parameters for processing
-                table_id, plan_id, plan_step, view_id = get_params(packets[0])
-
-                # Run the pipeline to create a dataset for the product
-                pipeline = CoDICEL1aPipeline(table_id, plan_id, plan_step, view_id)
-                pipeline.get_esa_sweep_values()
-                pipeline.get_acquisition_times()
-                pipeline.get_data_products(apid)
-                pipeline.unpack_science_data(science_values)
-                dataset = pipeline.create_science_dataset(met, data_version)
-
-    # TODO: Temporary workaround in order to create hi data products in absence
-    #       of simulated data. This is essentially the same process as is for
-    #       lo, but don't try to decom any packets, just define the data
-    #       outright.
-    elif file_path.name.startswith("imap_codice_l0_hi"):
-        if file_path.name.startswith("imap_codice_l0_hi-counters-aggregated"):
-            apid = CODICEAPID.COD_HI_INST_COUNTS_AGGREGATED
-            table_id, plan_id, plan_step, view_id = (1, 0, 0, 3)
-        elif file_path.name.startswith("imap_codice_l0_hi-counters-singles"):
-            apid = CODICEAPID.COD_HI_INST_COUNTS_SINGLES
-            table_id, plan_id, plan_step, view_id = (1, 0, 0, 4)
-        elif file_path.name.startswith("imap_codice_l0_hi-omni"):
-            apid = CODICEAPID.COD_HI_OMNI_SPECIES_COUNTS
-            table_id, plan_id, plan_step, view_id = (1, 0, 0, 5)
-        elif file_path.name.startswith("imap_codice_l0_hi-sectored"):
-            apid = CODICEAPID.COD_HI_SECT_SPECIES_COUNTS
-            table_id, plan_id, plan_step, view_id = (1, 0, 0, 6)
-
-        met0 = (np.datetime64("2024-04-29T00:00") - IMAP_EPOCH).astype("timedelta64[s]")
-        met0 = met0.astype(np.int64)
-        met = [met0, met0 + 1]  # Using this to match the other data products
-        science_values = ""  # Currently don't have simulated data for this
-
-        pipeline = CoDICEL1aPipeline(table_id, plan_id, plan_step, view_id)
-        pipeline.get_data_products(apid)
-        pipeline.unpack_science_data(science_values)
-        dataset = pipeline.create_science_dataset(met, data_version)
-
-    # Write dataset to CDF
-    logger.info(f"\nFinal data product:\n{dataset}\n")
+    # TODO: Use new packet_file_to_dataset() function to simplify things
+
+    # Decom the packets, group data by APID, and sort by time
+    packets = decom_packets(file_path)
```

Comment:
This is where you could use the function that Greg just added `packet_file_to_dataset()`. That would require some modification of your `create_dataset` functions though.

Reply:
Aiming to tackle this in my next PR